React/src/components/DynamicEditCellComponent.tsx

```diff
@@ -0,0 +1,83 @@
+import NumberBox from "devextreme-react/number-box";
+import TextBox from "devextreme-react/text-box";
+import DateBox from "devextreme-react/date-box";
+import CheckBox from "devextreme-react/check-box";
+import SelectBox from "devextreme-react/select-box";
+
+import { ValueChangedEvent as NumberValueChanged } from "devextreme/ui/number_box";
+import { ValueChangedEvent as TextValueChanged } from "devextreme/ui/text_box";
+import { ValueChangedEvent as DateValueChanged } from "devextreme/ui/date_box";
+import { ValueChangedEvent as CheckValueChanged } from "devextreme/ui/check_box";
+
+import { ValueChangedEvent as SelectValueChanged } from "devextreme/ui/select_box";
+
+import DataSource from "devextreme/data/data_source";
+import service from "../data";
+import { useCallback } from "react";
+import { ColumnEditCellTemplateData } from "devextreme/ui/data_grid";
+
+const dataSource = new DataSource({
+    store: {
+      type: "array",
+      key: "ID",
+      data: service.getDropDownData(),
+    }
+});
+
+function DynamicEditCellComponent({ data }: { data: ColumnEditCellTemplateData }){
```

Comment:
ColumnEditCellTemplateData also can be like ColumnEditCellTemplateData<SampleItem, Number>